packages/cardano-services-client/src/HttpProvider.ts

```diff
@@ -78,11 +78,9 @@ export const createHttpProvider = <T extends Provider>({
   new Proxy<T>({} as T, {
     // eslint-disable-next-line sonarjs/cognitive-complexity
     get(_, prop) {
-      if (prop === 'then') return;
```

Comment:
why is this not needed anymore? I suppose it was handling waiting for a promise to resolve

Reply:
oh, got it. It's going to be handled by the missing prop code path

Reply:
A couple lines below this one does the job, this is redundant now that it doesn't throw (returns undefined instead)